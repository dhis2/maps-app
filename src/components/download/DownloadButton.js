import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import i18n from '@dhis2/d2-i18n';
import { withStyles } from '@material-ui/core/styles';
import { Button } from '@material-ui/core';
import DownloadDialog from './DownloadDialog';
import { toggleDownloadDialog } from '../../actions/download';

const styles = {
    label: {
        textTransform: 'none',
        fontSize: 15,
        fontWeight: 400,
    },
};

export class DownloadButton extends Component {
    static propTypes = {
        toggleDownloadDialog: PropTypes.func.isRequired,
        classes: PropTypes.object.isRequired,
    };

    render() {
        const { classes } = this.props;

        return (
            <Fragment>
                <Button
                    key="button"
                    onClick={this.onClick}
                    classes={{
                        label: classes.label,
                    }}
                >
                    {i18n.t('Download')}
                </Button>
                <DownloadDialog />
            </Fragment>
        );
    }

    onClick = () => this.props.toggleDownloadDialog(true);
}

export default connect(
    null,
    { toggleDownloadDialog }
)(withStyles(styles)(DownloadButton));
