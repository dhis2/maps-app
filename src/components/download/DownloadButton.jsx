import i18n from '@dhis2/d2-i18n'
import cx from 'classnames'
import React from 'react'
import { openDownloadMode } from '../../util/history.js'
import styles from './styles/DownloadButton.module.css'

const DownloadButton = () => {
    return (
        <button
            className={cx(
                styles.button,
                'push-analytics-download-dropdown-menu-button'
            )}
            onClick={openDownloadMode}
        >
            {i18n.t('Download')}
        </button>
    )
}

export default DownloadButton
