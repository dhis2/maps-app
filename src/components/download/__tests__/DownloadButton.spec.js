import React from 'react';
import { shallow } from 'enzyme';
import { DownloadButton } from '../DownloadButton';

describe('DownloadButton', () => {
    const renderComponent = props =>
        shallow(
            <DownloadButton
                classes={{}}
                toggleDownloadDialog={() => null}
                {...props}
            />
        );

    it('Should render a download button', () => {
        const wrapper = renderComponent();
        expect(wrapper.find('MenuButton').exists()).toBe(true);
        expect(
            wrapper
                .find('MenuButton')
                .render()
                .text()
        ).toEqual('Download');
    });

    it('Should render a download dialog', () => {
        const wrapper = renderComponent();
        expect(wrapper.find('Connect(DownloadDialog)').exists()).toBe(true);
    });

    it('Should render a download button and a dialog component', () => {
        const wrapper = renderComponent();
        expect(wrapper).toMatchSnapshot();
    });

    it('should call toggleDownloadDialog when download button is clicked', () => {
        const toggleDownloadDialogSpy = jest.fn();
        const wrapper = renderComponent({
            toggleDownloadDialog: toggleDownloadDialogSpy,
        });

        wrapper.find('MenuButton').simulate('click', true);
        expect(toggleDownloadDialogSpy).toHaveBeenCalled();
    });
});
