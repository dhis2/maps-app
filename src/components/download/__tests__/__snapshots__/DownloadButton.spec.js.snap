// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadButton Should render a download button and a dialog component 1`] = `
<Fragment>
  <MenuButton
    onClick={[Function]}
  >
    Download
  </MenuButton>
  <Connect(DownloadDialog) />
</Fragment>
`;
